### PR TITLE
core: Migration to Next v15

### PR DESCRIPTION
**Pull request by pingSubhajit**
### Issue:
[LET-21: core: Migrate to Next JS v15](https://linear.app/letraz/issue/LET-21/core-migrate-to-next-js-v15)

### Description:
This PR updates the Letraz application to Next.js v15, taking advantage of the latest features, performance improvements, and security enhancements. The migration follows the official Next.js v15 migration guide and utilizes the provided codemod tool for a smooth transition.

### Implemented Changes:

- Updated the Next.js dependency in the `package.json` file to version 15.
- Adjusted configurations and code based on Next.js v15 documentation.
- Migrated custom middleware and API routes to the new Next.js API routes framework.
- Thoroughly tested the application after migration to ensure functionality and performance.
- Implemented type assertions as a temporary fix for broken types in the Framer Motion library due to changes in Next.js v15.

### Closing Note:
This migration successfully updates the Letraz application to Next.js v15, leveraging the latest capabilities and ensuring the application remains up-to-date. The temporary type assertions for Framer Motion will be revisited once the library's types are updated to be compatible with Next.js v15.

### CODE REVIEW

**Review by sourabhrathourr — APPROVED** [review]
This migration seems to be working fine for me and follows all the guidelines listed by next js migration doc, there are no breaking or minor issues as such.  We should merge it to main branch.

File: components/animations/TextAnimations.tsx [hunk 1]
```diff
@@ -229,13 +229,16 @@ const TextAnimate: FC<Props> = ({
 	 */
 
 	if (type === 'rollIn' || type === 'whipIn') {
-		return ( // @ts-ignore
-			<h2 {...props}>
+		return (
+			// @ts-ignore
```
Comment (sahilverma-dev) [inline components/animations/TextAnimations.tsx:233]: Should we include the reason for this `ts-ignore` @pingSubhajit

**Review by sahilverma-dev — CHANGES_REQUESTED** [review]
Some manual adjustments of indentation for types may be needed to ensure the code follows the project guidelines.

File: components/animations/TextAnimations.tsx (hunk 1 above)
Comment (pingSubhajit) [reply components/animations/TextAnimations.tsx]: Agreed. This was a minor oversight that is now fixed using type assertion. Thanks for the catch.

**Pull request merged**